fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/MultiJoinNode.java

```diff
@@ -59,6 +62,22 @@ private static void flattenJoinNode(OptExpression node, LinkedHashSet<OptExpress
             return;
         }
 
+        if (joinOperator.getProjection() != null) {
+            Projection projection = joinOperator.getProjection();
+
+            for (Map.Entry<ColumnRefOperator, ScalarOperator> entry : projection.getColumnRefMap().entrySet()) {
+                if (!entry.getValue().isColumnRef()) {
+                    atoms.add(node);
+                    return;
+                }
+
+                if (!entry.getKey().equals(entry.getValue())) {
+                    atoms.add(node);
+                    return;
+                }
+            }
+        }
```

Comment:
Why can't support? According to the current practice, the projection should not appear on the join node

Reply:
When this PR has an expression directly in the JOIN, the original logic is used, and no join reorder is performed